[ML] Adds Pattern analysis embeddable for dashboards

jgowdyelastic opened the pull request:
Adds a new embeddable version of the pattern analysis component.

![image](https://github.com/user-attachments/assets/984921b5-cd33-4f7f-9207-c3182c97d015)

The options to configure it are the same as the menu available in the Patterns tab in Discover.

![image](https://github.com/user-attachments/assets/33e3b537-c4e2-475b-89ad-8fdfef347fdc)

**Actions**
The user can choose to filter a pattern in or out in in the current dashboard, or jump to Discover.
![image](https://github.com/user-attachments/assets/dfd9dc3d-932c-463b-8ea5-e93fa6caf518)

joana-cps commented:
## Design Review

- [ ] **Change to Overlay flyout**
Since this panel don't have a preview available we should use the Overlay flyout
<img width="624" alt="Screenshot 2024-07-12 at 12 46 38" src="https://github.com/user-attachments/assets/b1137a3d-6d1b-4ff6-98d2-03b107ac4722">

>

Here are the settings that should be used in the Header, Body and Footer of the Overlay flyout:
`<EuiFlyout size="m" paddingSize="m">` 
<img width="985" alt="Screenshot 2024-07-12 at 13 12 21" src="https://github.com/user-attachments/assets/2570668c-7811-4d4c-9a48-f2a3e907da75">
> [!NOTE]  
> Advised `Overlay flyout size` is being discussed at the moment by Dashboard team. This can probably change to a size="s" or fixed(500)



### Enhancements
- [ ] **10px content padding**
For consistency's sake, we should have the same content padding as the other panels.
<img width="554" alt="Screenshot 2024-07-12 at 13 01 16" src="https://github.com/user-attachments/assets/bfca9da2-3059-4a9f-8e98-52a1a1a98fb0">

⌛ more comments to be added...

jgowdyelastic commented:
@joana-cps thanks for the comments.
I've stuck with a small flyout size as medium looked odd with the super wide dropdown menus.
Regarding using `compressed` for every form item, is this mandatory? IMO the non-compressed versions look better when the form does not contain many items. Especially the data view selector.

![image](https://github.com/user-attachments/assets/3a0406de-8ed0-4b08-8e65-59ef73137829)

vs

![image](https://github.com/user-attachments/assets/02f031a9-e391-4683-908a-b820d100ad2c)

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

jgowdyelastic commented on x-pack/plugins/aiops/public/embeddables/pattern_analysis/pattern_analysis_initializer.tsx:
```diff
@@ -0,0 +1,300 @@
+            >
+              <FormattedMessage
+                id="xpack.aiops.embeddablePatternAnalysis.config.applyAndCloseLabel"
+                defaultMessage="Save"
```
The style guideline posted above states that this should be `Save`
https://github.com/elastic/kibana/pull/186539#issuecomment-2225371151

peteharverson replied:
Please go with these [guidelines](https://github.com/elastic/kibana/pull/184030#issuecomment-2183508806) from Andrea so it's consistent with the other dashboard panels. There is an inconsistency in the EUI guidelines which we'll get updated.

peteharverson commented:
I'm seeing some occasional odd effects when using Filter in / Filter out:

- Do one filter for action followed by a filter out - end up with two filters. Then quickly remove both filters - view refreshes to show the results as if the first 'filter for' was still present rather than the complete list of patterns:

https://github.com/user-attachments/assets/a3ceb1cd-3839-45f4-9622-f4bf50a463dd

- Do one filter for action followed by a filter out - sometimes can result in a 'Bad request' error (possibly when filtering the same pattern in and out?)

andreadelrio commented:
> @joana-cps thanks for the comments. I've stuck with a small flyout size as medium looked odd with the super wide dropdown menus. Regarding using `compressed` for every form item, is this mandatory? IMO the non-compressed versions look better when the form contains only a few items.

Hi @jgowdyelastic, as mentioned over Slack I'll be taking over the design review of this given Joana is on leave. 

- Re: `compressed` items - yes, we're strongly recommending this for all flyouts in Dashboards. Any flyout which doesn't currently use `compressed` forms will be updated to do so. The guidelines that Joana screenshotted above are not from EUI. It is a [document](https://www.figma.com/design/y6thIbIHWSXl3v2GzOTtrE/%5BWIP%5D-Dashboard-panel-guidelines?node-id=0-1&t=4BlWsjJ8wOfDSrwE-1) that I'm working on to provide UX guidance from the Presentation Team to other teams wanting **to add new embeddables** to Dashboard. It is still not 100% finalized but I think it's worth checking it out and bookmarking. The ultimate goal is to achieve consistency and cleanliness in all Dashboard editors regardless of the teams implementing them. We welcome feedback. cc @peteharverson 

- Regarding the main call to action, `Apply and close` vs `Save`, I would like to clarify something. In the `Field Statistics` [flyout](https://github.com/elastic/kibana/pull/184030#issuecomment-2183508806) we use `Apply and close` because a) there's a `preview` and (therefore) b) the flyout is a `push` flyout. A `push` flyout allows users to see a preview of their panel while they change settings in the editor. In the current implementation of the `Pattern analysis` embeddable, I don't see a `Preview` available which is the reason Joana was suggesting an `overlay` flyout (which, in turn, would use the `Save` button). My question is, is it possible to provide a preview for the `Pattern analysis` pattern? That way we could use the push flyout, like this: 

<img width="1440" alt="Frame 2" src="https://github.com/user-attachments/assets/62fb9c92-4285-4c91-98e7-6103b1067fcb">

jgowdyelastic commented:
@andreadelrio I've [updated](https://github.com/elastic/kibana/pull/186539/commits/7667e04bf10696502b81fd4ca7297fb60e00f070) the editor to show a live preview.
I've also [switched](https://github.com/elastic/kibana/pull/186539/commits/41968c00fb562d0739589e119bdcb830945c764e) to compressed components.

andreadelrio reviewed (changes requested):
Thanks for the changes @jgowdyelastic, this is looking good!

A couple of comments:

- Why are you using a **EuiComboBox** for `Data view` when you allow for the selection of only one at a time?

![CleanShot 2024-07-24 at 15 45 18](https://github.com/user-attachments/assets/8318fa5a-0f60-4d45-888c-2810d18f941c)

- Can we use something like what we use for `Controls` instead?
<img width="416" alt="image" src="https://github.com/user-attachments/assets/d99ffe8a-305f-4aa7-8ab2-6d15e9397907">

- Can you use size 's' for this **EuiCallout** and pass the `alert` symbol to the title?
<img width="535" alt="image" src="https://github.com/user-attachments/assets/d9a108ff-90cc-4345-8671-269ae2a6f182">

End result:
<img width="540" alt="image" src="https://github.com/user-attachments/assets/1ebdef62-c147-43f1-ad47-ecae7fa2ab68">

- I think I might have found a bug. I select a data view where I know pattern analysis is available (Logs > message). I see the preview and it looks ok. I then select another data view (where I'm guessing pattern analysis is not available) so I get the warning callout. When I go back to select the data view that was working a second ago (Logs), it produces the warning callout instead of the preview. I select a third data view and then go back to select the first data view (i.e. Logs) and inexplicably it works again. Check this out:

![CleanShot 2024-07-24 at 15 51 19](https://github.com/user-attachments/assets/b5fb4a26-022b-464b-aa47-b1878f556e5c)

jgowdyelastic commented:
@andreadelrio 
>Why are you using a EuiComboBox for Data view when you allow for the selection of only one at a time?

This is a [data view selector](https://github.com/elastic/kibana/blob/main/src/plugins/unified_search/public/index_pattern_select/index_pattern_select.tsx) shared from the unified search plugin, we also use it in the change point detection embeddable. I agree that it would be better as single select. I can see if this can be changed (but outside of the scope of this PR)

>Can you use size 's' for this EuiCallout and pass the alert symbol to the title?

Changed in https://github.com/elastic/kibana/pull/186539/commits/3a7c0393dea492f2738e8b40ecc00d902a0da5d5

>I think I might have found a bug.

Yeah this was a bug. Fixed in https://github.com/elastic/kibana/pull/186539/commits/3a7c0393dea492f2738e8b40ecc00d902a0da5d5

peteharverson commented:
Seems to be an issue if you apply a custom time range to the panel, where the patterns don't get rendered as expected when the Dashboard time range doesn't include the time range set on the panel. I see the loading indicator for a few seconds but then it refreshes empty (or sometimes with the callout indicating the field isn't suitable for analysis).

<img width="2448" alt="Screenshot 2024-07-26 at 16 10 40" src="https://github.com/user-attachments/assets/34f2f4d1-1631-4b9d-8cfa-3f344803e0bd">

andreadelrio commented on x-pack/plugins/aiops/public/ui_actions/create_pattern_analysis_action.tsx:
```diff
@@ -0,0 +1,97 @@
+        getIconType: () => 'machineLearningApp',
+      },
+    ],
+    getIconType: () => 'machineLearningApp',
```
Please use the dedicated icon Joana created for this, `logPatternAnalysis`
```suggestion
    getIconType: () => 'logPatternAnalysis',
```

jgowdyelastic replied:
Updated in 09fe9b8c102582d77cd7a522ea890869360b6dd2

andreadelrio commented:
> > Why are you using a EuiComboBox for Data view when you allow for the selection of only one at a time?
> 
> This is a [data view selector](https://github.com/elastic/kibana/blob/main/src/plugins/unified_search/public/index_pattern_select/index_pattern_select.tsx) shared from the unified search plugin, we also use it in the change point detection embeddable. I agree that it would be better as single select. I can see if this can be changed (but outside of the scope of this PR)

@stratoula or @mbondyra can either of you tell me where we're using this data view selector powered by **EuiComboBox**?

andreadelrio commented:
@jgowdyelastic Thanks for the changes. Upon testing I think I might have ran into another bug. I'm testing using sample data (Logs and Flights). When I select Logs I see the field list. When I then switch to Flights the field list won't populate.

![CleanShot 2024-07-28 at 19 22 12](https://github.com/user-attachments/assets/a7e8eb2d-c3b9-4218-8609-e540160a4c11)

stratoula commented:
@andreadelrio no we are not using EUICombobox, we are using a button with a popover. The menu was very complicated and could not use eui combobox to create this

jgowdyelastic commented:
> @jgowdyelastic Thanks for the changes. Upon testing I think I might have ran into another bug. I'm testing using sample data (Logs and Flights). When I select Logs I see the field list. When I then switch to Flights the field list won't populate.

The flights data set does not contain any text fields for selection.
I've added a callout to inform the user in this scenario.
![image](https://github.com/user-attachments/assets/4c49808f-c172-4c67-97f3-3438116b9af3)

jgowdyelastic commented:
> Seems to be an issue if you apply a custom time range to the panel, where the patterns don't get rendered as expected when the Dashboard time range doesn't include the time range set on the panel. I see the loading indicator for a few seconds but then it refreshes empty (or sometimes with the callout indicating the field isn't suitable for analysis).
> 

@peteharverson fixed in 3cee21f2671bb63424f3aa79c25eb143a6fa96c8

andreadelrio commented:
> @andreadelrio but we use EuiCombobox for dataview in the filter popover, but it only appears in Lens if there are multiple layers on different dataviews: <img alt="Screenshot 2024-07-29 at 14 42 28" width="1724" src="https://private-user-images.githubusercontent.com/4283304/353063871-576956be-3a8f-4ff7-ae70-e8136e4a6ad5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNTczNDksIm5iZiI6MTcyMjI1NzA0OSwicGF0aCI6Ii80MjgzMzA0LzM1MzA2Mzg3MS01NzY5NTZiZS0zYThmLTRmZjctYWU3MC1lODEzNmU0YTZhZDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjlUMTI0NDA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWEzZDA5NDdkZTZjNjZhYTY0NGM2MTY1NDgwMGQ2ZTZkYTIyZjg0NTIwY2QxMTZlODdhMDQ4OGI4ZmFmMDgwNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.i8wHA89Z3LeK5i88Yay-8LdqVHkdNZ5BJTWLguvOMAI">

Thanks @mbondyra. What I'm trying to understand is why if this is the same component, @jgowdyelastic is getting the box around the data view name and the clear button. 

<img width="409" alt="Frame 1321315066" src="https://github.com/user-attachments/assets/e09c4874-fdb1-4abd-9bae-ff64bf8eab7c">

![CleanShot 2024-07-29 at 17 57 37](https://github.com/user-attachments/assets/3f3db5e0-f9c8-4f31-98a7-2920f70edc5f)


vs the component in the `Add filter` popover

![CleanShot 2024-07-29 at 17 54 47](https://github.com/user-attachments/assets/8bcad0de-1c6a-4f2e-ae0d-a6daed48bae4)

@jgowdyelastic I don't mind the **EuiComboBox** but if it could behave like the `Add filter Data view selector` it would be much better. Let me know if this is doable.

mbondyra commented:
@andreadelrio I checked and it's not the same component – it's definitely something we could improve from Unified Search side. The component we use in the filter is [here](https://github.com/elastic/kibana/blob/main/src/plugins/unified_search/public/filter_bar/filter_editor/filter_editor.tsx#L342). 
@jgowdyelastic I submitted a PR to render as a text and not pills: https://github.com/elastic/kibana/pull/189474 Once it's merged it should be displayed as expected.

peteharverson commented on x-pack/plugins/aiops/public/ui_actions/create_pattern_analysis_action.tsx:
```diff
@@ -0,0 +1,97 @@
+    getIconType: () => 'logPatternAnalysis',
+    getDisplayName: () =>
+      i18n.translate('xpack.aiops.embeddablePatternAnalysisDisplayName', {
+        defaultMessage: 'Pattern analysis',
```
Is this text being used to generate the text for the dashboard panel actions? The other embeddables seem to use sentence case e.g. `Edit swim lane`. Can we have `Edit pattern analysis` in here?

<img width="353" alt="Screenshot 2024-07-30 at 12 41 22" src="https://github.com/user-attachments/assets/f8e676ac-53ac-442f-9165-73fd08116c83">

jgowdyelastic replied:
Updated in https://github.com/elastic/kibana/pull/186539/commits/30abc391afd55e5b897079b205e26bc994fef37b

peteharverson commented:
@andreadelrio @mbondyra do you know why the data views in the data view selector aren't sorted alphabetically? I know you can start typing text to search for a specific data view, but without sorting it is difficult to find the one of interest when you have more than a few data views.

<img width="621" alt="Screenshot 2024-07-30 at 12 53 01" src="https://github.com/user-attachments/assets/21aa4a71-7b57-41d2-ac72-b0823dbb7e69">

peteharverson reviewed (approved):
Tested latest changes and all filtering / data view / mesage field actions work as expected.

Other than the comment around the text for the edit action, LGTM

mbondyra commented:
> do you know why the data views in the data view selector aren't sorted alphabetically? 

Added sorting to my PR above.

darnautov commented:
When I change the **Minimum time range** during the embeddable initialization, I see 3  outgoing bsearch requests and 3 of them seem identical    
<img width="513" alt="image" src="https://github.com/user-attachments/assets/1a922473-db1a-4dfd-a584-cd3fca0c8aa9">

darnautov commented:
This one is easier to reproduce with network throttling. When I click "Cancel" after updating embeddable configuration, the panel turns blank and never recovers. 


https://github.com/user-attachments/assets/4be3ffd4-0a04-4f05-8508-067d96930b0b

darnautov commented on x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/log_categorization_for_discover.tsx:
```diff
@@ -0,0 +1,472 @@
+        });
+      }
+    } catch (error) {
+      if (error.name === 'AbortError') {
+        // ignore error
+      } else {
+        toasts.addError(error, {
+          title: i18n.translate('xpack.aiops.logCategorization.errorLoadingCategories', {
+            defaultMessage: 'Error loading categories',
+          }),
+        });
+      }
+    }
```
nit: 
```suggestion
      if (error.name !== 'AbortError') {
        toasts.addError(error, {
          title: i18n.translate('xpack.aiops.logCategorization.errorLoadingCategories', {
            defaultMessage: 'Error loading categories',
          }),
        });
      }
    }
```

jgowdyelastic replied:
Updated in https://github.com/elastic/kibana/pull/186539/commits/a3008dc3d0f23cb78be2899bcb59236eec21d846

darnautov commented on x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/log_categorization_for_discover.tsx:
```diff
@@ -0,0 +1,472 @@
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [input.lastReloadRequestTime]
```
can we list the complete set of deps?

jgowdyelastic replied:
Updated in https://github.com/elastic/kibana/pull/186539/commits/a3008dc3d0f23cb78be2899bcb59236eec21d846

darnautov commented on x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/log_categorization_for_embeddable.tsx:
```diff
@@ -332,140 +285,94 @@ export const LogCategorizationEmbeddable: FC<LogCategorizationEmbeddableProps> =
+      if (typeof onChange === 'function') {
+        onChange(data?.categories ?? []);
+      }
+      if (data !== null) {
```
Defined data doesn't always imply that the visualisation has been rendered. Did you test the PDF report?

jgowdyelastic replied:
This is a good spot. I didn't realise it was being used in that way and indeed the PDF export is catching the component too early.

jgowdyelastic replied:
I've removed this and made changes to improve the reporting of the rendering being complete in https://github.com/elastic/kibana/pull/186539/commits/be39b67b5884076c5fba701574af17572423dede
However it isn't catching the chart being rendered, only the loading message. I've tested a few other embeddables and they are also doing this, so I think there is a deeper problem with the PDF export.

darnautov commented on x-pack/plugins/aiops/public/embeddables/pattern_analysis/pattern_analysis_initializer.tsx:
```diff
@@ -0,0 +1,402 @@
+  const [formInput, setFormInput] = useState<PatternAnalysisEmbeddableRuntimeState>(
+    pick(
+      initialInput ?? {
+        minimumTimeRangeOption: '1 week',
```
The options have to have i18n support

jgowdyelastic replied:
This is a good spot. I don't know how this got missed before, but all of these options are [not translated](https://github.com/elastic/kibana/blob/be39b67b5884076c5fba701574af17572423dede/x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/minimum_time_range.ts).
I'll fix this for this PR

jgowdyelastic replied:
Updated in bcfe91f2aeeb2807cd613c50e70184f852e7e181

darnautov commented on x-pack/plugins/aiops/public/embeddables/pattern_analysis/pattern_analysis_initializer.tsx:
```diff
@@ -0,0 +1,402 @@
+    function initFields() {
+      dataViews.get(dataViewId).then((dataView) => {
+        const isTimeBased = dataView.isTimeBased();
+        setIsDataViewTimeBased(isTimeBased);
```
can potentially set state for unmounted component

jgowdyelastic replied:
Updated in https://github.com/elastic/kibana/pull/186539/commits/a3008dc3d0f23cb78be2899bcb59236eec21d846

darnautov commented on x-pack/plugins/aiops/public/embeddables/pattern_analysis/pattern_analysys_component_wrapper.tsx:
```diff
@@ -0,0 +1,77 @@
+      }}
+    >
+      <LogCategorizationEmbeddable
+        input={{
```
With the new embeddable framework, it's expected from us to get rid of the `input` concept

jgowdyelastic replied:
Updated in https://github.com/elastic/kibana/pull/186539/commits/be39b67b5884076c5fba701574af17572423dede

darnautov commented on x-pack/plugins/aiops/public/shared_components/pattern_analysis.tsx:
```diff
@@ -0,0 +1,172 @@
+      data-test-subj="aiopsEmbeddablePatternAnalysis"
+      css={{
+        width: '100%',
+        padding: '10px',
```
nit, can we use eui padding instead?

jgowdyelastic replied:
I cannot find an appropriate eui size for `10px` and I have been asked in a comment above to use `10px` to match the padding of other embeddables.

jgowdyelastic commented on x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/log_categorization_for_discover.tsx:
```diff
@@ -0,0 +1,472 @@
+      if (typeof input.switchToDocumentView === 'function') {
+        input.switchToDocumentView();
+      }
+      setTimeout(() => {
+        filterManager.addFilters([filter]);
+      }, 500);
```
It's a work around to solve a recent bug where switching to the documents tab would fail. The filters need to be added slightly after the call to switch tabs.
I appreciate using a delay like this is a bit hacky, but I could not find a better solution, from my testing this does not fail.

darnautov replied:
Discover's appState has a `subscribe` method. You can try subscribing to a state change and adding filters on the `viewMode` update.

jgowdyelastic replied:
Good idea. Updated in https://github.com/elastic/kibana/pull/186539/commits/033e8e3b3af507c64c5886c3105cac251218f6e6

jgowdyelastic commented on x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/log_categorization_for_discover.tsx:
```diff
@@ -0,0 +1,472 @@
+  } = useMinimumTimeRange();
+  const { filters, query } = useMemo(() => getState(), [getState]);
+
+  const mounted = useRef(false);
```
Updated in https://github.com/elastic/kibana/pull/186539/commits/a3008dc3d0f23cb78be2899bcb59236eec21d846

andreadelrio reviewed (approved):
Design changes LGTM, tested locally. Thanks @jgowdyelastic and thanks to @mbondyra for her support with the data view selector.

jughosta commented:
Hi @jgowdyelastic,

Great that we are introducing Pattern Analysis embeddable!

I started reviewing the PR and noticed some issues on Discover page:

- Switching to "Patterns" tab adds spacing above the tabs. Can we avoid this jump? I would expect any Patterns content to go below the tabs. Also the field selector and actions layout looks a bit off to me. Shouldn't they be on the same line? cc @andreadelrio 

![Aug-08-2024 15-29-05](https://github.com/user-attachments/assets/d7b818b0-ff5a-451b-be21-67ba261fb560)

- Updating a query in Discover search input triggers endless reloading of Patterns tab

![Aug-08-2024 15-31-29](https://github.com/user-attachments/assets/27900190-8bc0-40f7-8471-37bc2c748fba)

Does it work differently for you?

jughosta commented:
> Also the field selector and actions layout looks a bit off to me. Shouldn't they be on the same line?

Could be because of https://github.com/elastic/kibana/issues/190247 
Looking into it.

jgowdyelastic commented:
@jughosta The layout issues have been fixed by https://github.com/elastic/kibana/pull/189883

jgowdyelastic commented:
> When I change the **Minimum time range** during the embeddable initialization, I see 3 outgoing bsearch requests and 3 of them seem identical 

@darnautov I'm not able to reproduce this. I may have fixed it when making other changes.

jgowdyelastic commented:
> This one is easier to reproduce with network throttling. When I click "Cancel" after updating embeddable configuration, the panel turns blank and never recovers.
> 
For some reason the info message was removed. I've added it back in now so the behaviour is the same as the Discover version. If the user presses cancel they'll see a message about no patterns being found.

jgowdyelastic commented on src/plugins/discover/public/application/main/components/layout/discover_main_content.tsx:
```diff
@@ -78,6 +78,17 @@ export const DiscoverMainContent = ({
         }
       }
+
+      return new Promise<VIEW_MODE>((resolve, reject) => {
```
This is used by the Patterns tab to know when it can apply the filters after switching to the Documents tab.
Without waiting for the switch, the filters are not applied.

jughosta reviewed (approved):
Data Discovery changes LGTM 👍 Thanks!

`*` as a query seem to trigger errors:
<img width="1822" alt="Screenshot 2024-08-19 at 16 36 33" src="https://github.com/user-attachments/assets/166f8383-ed02-4aa8-a385-78365ac07a81">
<img width="1618" alt="Screenshot 2024-08-19 at 16 40 27" src="https://github.com/user-attachments/assets/fef29163-8fbd-4a37-bb3a-d9e7264a703b">

jgowdyelastic commented:
> `*` as a query seem to trigger errors: <img alt="Screenshot 2024-08-19 at 16 36 33" width="1822" 

Thanks @jughosta. This problem appears on main too. I've raised an [issue](https://github.com/elastic/kibana/issues/190710)

darnautov reviewed (approved):
LGTM
